api/osf_api.py

```diff
@@ -401,3 +401,59 @@ def get_preprint_downloads_count(session=None, node_id=None):
         return data[0]['attributes']['extra']['downloads']
     else:
         return None
+
+
+def get_registration_schemas_for_provider(
+    session=None, provider_id='osf', data_type='name'
+):
+    """Returns a list of allowed registration schemas for an individual provider.  The
+    list will be either schema names or a paired list of schema names and ids.  The
+    default data type is 'name'.  The default provider_id is 'osf'.
+    """
+    if not session:
+        session = get_default_session()
+    url = 'v2/providers/registrations/{}/schemas/'.format(provider_id)
+    # NOTE: Using '50' as the page size query parameter here. We don't actually have 50
+    # total registration schemas. It's under 30 at this time, but using 50 here gives us
+    # plenty of room to add more schemas without having to update this function.
+    data = session.get(url, query_parameters={'page[size]': 50})['data']
+    if data:
+        schema_list = []
+        for schema in data:
+            name = schema['attributes']['name']
+            if data_type == 'name':
+                schema_list.append(name)
+            elif data_type == 'name_id':
+                schema_id = schema['id']
+                schema_list.append([name, schema_id])
+        return schema_list
```

Comment:
*Possible* golf-pportunity:

```
return [[s['attributes']['name'], s.get('id', None)] for s in data]
```

Or if `id` is always `None` for `name` schemas:

```
return [[s['attributes']['name'], s['id']] for s in data]
```

Reply:
Yeah. I simplified this function by always returning the (name, id) pair list.